### DISCUSSION TITLE
ChatOps and MS Teams "resource could not be found" error

### DISCUSSION THREAD

**Discussion by rdancel001** [StackStorm/st2#5773] [ChatOps]
Hey all, Im trying to integrate MS Teams and ChatOps. Im at the point where it's configured and I am able to send messages to the bot via Teams, its able to get to st2 via nginx, but when the requests get to the bot it returns a 404. I'm testing via the bot framework portal (https://dev.botframework.com) and using the test panel on the side. In the test window I'm just typing "help", which should return the list of available commands, to verify connection and it can respond back.

Has anyone here gotten it to work properly? 

Here's what I get from nginx and st2api logs
**nginx logs:**
> [13/Oct/2022:18:17:05 -0400] "POST /api/messages HTTP/1.1" 404 50 "-" "BF-DirectLine (Microsoft-BotFramework/3.2 +https://botframework.com/ua)"
So nginx got the request.

**st2api.log**
> 2022-10-13 18:17:03,202 140342904908384 INFO logging [-] 4ec228ce-b299-4d40-9ce6-65adb2923bc9 - POST /messages with query={} (method='POST',path='/messages',remote_addr='127.0.0.1',query={},request_id='4ec228ce-b299-4d40-9ce6-65adb2923bc9')
2022-10-13 18:17:03,203 140342904908384 INFO logging [-] 4ec228ce-b299-4d40-9ce6-65adb2923bc9 - 404 50 0.903ms (method='POST',path='/messages',remote_addr='127.0.0.1',status=404,runtime=0.903,content_length=50,request_id='4ec228ce-b299-4d40-9ce6-65adb2923bc9')

**Running a Curl cmd equivalent**
> curl -X POST -H "Content-Type: application/json" -d '{"test":"test"}' https://st2.example.com/api/messages
> {"faultstring":"The resource could not be found."}

**Comment by cyberkov** [comment]
It looks like you connected teams directly to stackstorm. Actually the bot is running on port 8080,so you would need to configure a proxy vhost in nginx connecting /messages to localhost:8080.
The bot will then communicate with your stackstorm instance.

Hope this helps 🙂
